src/Client.php

```diff
@@ -267,7 +268,14 @@ private function configureDefaults(array $config)
         ];
 
         // idn_to_ascii() is a part of ext-intl and might be not available
-        $defaults['idn_conversion'] = function_exists('idn_to_ascii');
+        $defaults['idn_conversion'] = function_exists('idn_to_ascii')
+            // Old ICU versions don't have this constant, so we are basically stuck (see https://github.com/guzzle/guzzle/pull/2424
+            // and https://github.com/guzzle/guzzle/issues/2448 for details)
+            && (
+                defined('INTL_IDNA_VARIANT_UTS46')
+                ||
+                version_compare(PHP_VERSION, '7.2.0') < 0
```

Comment:
This would be faster:

```suggestion
                PHP_VERSION_ID < 70200
```